packages/wrangler/src/r2/helpers.ts

```diff
@@ -299,3 +299,15 @@ export async function putR2Sippy(
 		{ method: "PUT", body: JSON.stringify(config), headers }
 	);
 }
+
+
+/**
+ * R2 bucket names must only contain alphanumeric and - characters.
+ */
+export function isValidR2BucketName(
+	name: string | undefined
+): name is string {
+	return (
+		typeof name === "string" && /^[a-zA-Z][a-zA-Z0-9-]*$/.test(name)
```

Comment:
Is this mentioned in the R2 documentation? If so, it'd be good to link in a comment

Reply:
Not currently, however I can request this to be added to the documentation too.

Reply:
https://github.com/cloudflare/cloudflare-docs/pull/14355